installinstallmacos.py

```diff
@@ -370,17 +379,31 @@ def main():
                                          product_info[product_id]['BUILD'],
                                          product_info[product_id]['title'])
 
-    answer = raw_input(
-        '\nChoose a product to download (1-%s): ' % len(product_info))
-    try:
-        index = int(answer) - 1
-        if index < 0:
-            raise ValueError
-        product_id = product_info.keys()[index]
-    except (ValueError, IndexError):
-        print 'Exiting.'
+    # Quit now if displaynonly argument supplied
+    if args.displayonly:
         exit(0)
 
+    # check for specified build if argument supplied
+    if args.build:
+        for index, product_id in enumerate(product_info):
+            if args.build == product_info[product_id]['BUILD']:
+                answer = index+1
+                print '# %s chosen.' % answer
+
+    try:
+        answer
```

Comment:
If your goal is non-interactive use of the script, I'd think if --build didn't match an available build you'd exit with an error.